src/test/java/com/google/sps/WebSocketHandlerTest.java

```diff
@@ -0,0 +1,172 @@
+package com.google.sps;
+
+import com.google.sps.WebSocketHandler;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collections;
+import javax.websocket.Session;
+
+import org.junit.Test;
+import org.junit.Before;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+import static org.mockito.Mockito.*;
+
+public class WebSocketHandlerTest {
+
+  Constructor constructor;
+  Field instance;
+  Field chatRoomMap;
+  WebSocketHandler obj;
+  Map<Long, List<Session>> actualMap;
+
+  static long TEST_ROOM_ID;
+  static Session TEST_SESSION_1;
+  static Session TEST_SESSION_2;
+
+  @Before
+  public void setUp() throws NoSuchFieldException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+    TEST_SESSION_1 = mock(Session.class);
+    TEST_SESSION_2 = mock(Session.class);
+    TEST_ROOM_ID = 1;
+
+    constructor = WebSocketHandler.class.getDeclaredConstructor();
```

Comment:
"Test code using real APIs - don’t define back doors only for tests." - Google's unit test practices.

By using reflection and calling the constructor, you're changing the way this class can be used. The goal of unit tests is to be as close to production behavior as possible. Since in production you cannot call the constructor directly, focus on testing the API's that other classes can call. You might have to call more than one method to verify a behavior, and this is totally ok.